apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -1587,6 +1595,13 @@ def _delete_path_if_last(self, context, host=None, segment=None):
             atenant_id = self._get_network_aci_tenant(context.network.current)
             network_id = context.network.current['id']
             epg_name = self.name_mapper.endpoint_group(context, network_id)
+            if self._is_nat_enabled_on_ext_net(context.network.current):
+                # PTG name is different
+                network = context.network.current
+                l3out_name = self.name_mapper.l3_out(
+                    context, network['id'],
+                    openstack_owner=network['tenant_id'])
+                epg_name = self._get_ext_epg_for_ext_net(l3out_name)
             self._delete_port_path(
                 context, atenant_id, epg_name,
                 self._get_network_app_profile(context.network.current),
```

Comment:
Should the correct bd_name be provided to _get_network_app_profile() here as well?